app/scripts/modules/core/src/pipeline/executions/executionGroup/ExecutionGroup.tsx

```diff
@@ -242,7 +242,7 @@ export class ExecutionGroup extends React.Component<IExecutionGroupProps, IExecu
         key={execution.id}
         execution={execution}
         application={this.props.application}
-        onRerun={this.rerunExecutionClicked}
+        onRerun={canTriggerPipelineManually ? this.rerunExecutionClicked : undefined}
```

Comment:
maybe default to a noop just in case?

Reply:
that was actually my initial thought as well; turns out the Execution component checks for truthiness so any func object causes the re-run button to render. `undefined` allows React to omit the entire prop and in turn hide the re-run button. Think it would be worth changing the `Execution` component and approaching it differently?

Reply:
I was thinking the same thing...but the `<Execution` looks at the `onRerun` prop to determine whether to show the button, so this is a reasonable approach IMO.

Reply:
👍 makes sense!